akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/Behaviors.scala

```diff
@@ -209,6 +209,17 @@ object Behaviors {
   def monitor[T](interceptMessageClass: Class[T], monitor: ActorRef[T], behavior: Behavior[T]): Behavior[T] =
     scaladsl.Behaviors.monitor(monitor, behavior)(ClassTag(interceptMessageClass))
 
+  /**
+   * Interpret message and signal using given [[akka.actor.typed.Behavior]]
+   *
+   * execute message or signal using given behavior and switched to returned behavior
+   * note: [[Behaviors.same]] will cause actor switching to the given behavior
+   * and [[Behaviors.unhandled]] will trigger the [[akka.actor.typed.scaladsl.ActorContext.onUnhandled]]
+   * then switch to the given behavior.
+   */
+  def forward[T](behavior: Behavior[T], ctx: ActorContext[T], msg: T): Behavior[T] =
```

Comment:
It might be confusing to name it forward since that already exists in classic actors and this is different.

Also, might be better to place the method in the ActorContext instead of a factory in Behviors?

Reply:
2) sure, is that wrap it into setup solve the problem?
1) this feature came up to me as the analogy of `forward` that exists in classic actor at first time, so i use that name without further consideration.
so idk if it's ok to have the same name in both `Behaviors` and `Behavior`, since in essence this is a thin wrapper for `Behavior.interpret`.
and if it's not ok, we should probably use something like `switch`/`switchTo`  or `pipe`/`pipeTo` or just `behave` i guess?

Reply:
2. If we keep it in `Behaviors` it would indeed be good to wrap it in `setup` to have it executed lazily, and then the context parameter isn't needed. However, here the intended purpose is to run it (delegate) to another Behavior when already processing a message. The receiving Behavior is already running. That's why I thought that kind of usage would be encouraged by having it in the ActorContext. A bit similar to `context.ask`.

1. `forward` in classic actor has rather different semantics in my opinion. It is for sending a message to another actor while preserving the origin `sender()`. It's the same as `otherRef.tell(msg, sender())`. Here it's about delegating message processing to another `Behavior` (still within the scope of the same actor and the same message processing thread, without going through the mailbox).

`pipeToSelf` is already taken. `switchTo` isn't too bad. `context.delegate(msg)` might also be ok?

Reply:
thanks for clarifying the difference, now i see it's just same on the surface but completely different implication under the hood, also `delegate` is a pretty adequate name.
now i moved everything into `ActorContext.delegate`.